### DISCUSSION TITLE
Why is Vercel passing in values to dynamic segments during build when using nextjs 13 app directory?

### DISCUSSION THREAD

**Discussion by ghost** [vercel/community#1456] [Help] [answered]
### Summary

I have a local project that works fine but fails only on Vercel production deployment. 

### /app/[slug]/page.tsx

`import { sanityItem } from "../../lib/sanity";

export default async function BlogPage({
  params,
}: {
  params: { slug: string };
}) {
  const rawItem = await sanityItem(params.slug);
  console.log(
    `Params are ${JSON.stringify(params)} and rawItem is ${JSON.stringify(
      rawItem
    )}`
  );
  return <div>{JSON.stringify(rawItem, null, 2)}</div>;
}
`
### Build logs

Vercel build logs show that it is calling my dynamic segment twice with (what I think?) must be silly data:

info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback
--
10:55:24.945 | info  - Creating an optimized production build...
10:55:28.619 | info  - Compiled successfully
10:55:28.623 | info  - Skipping validation of types
10:55:28.623 | info  - Skipping linting
10:55:29.007 | info  - Collecting page data...
10:55:29.424 | info  - Generating static pages (0/4)
10:55:29.521 | info  - Generating static pages (1/4)
10:55:29.548 | info  - Generating static pages (2/4)
10:55:29.791 | Params are {"slug":"%5Bslug%5D"} and rawItem is undefined
10:55:30.015 | Params are {"slug":"%5Bslug%5D"} and rawItem is undefined

### Problem

This is breaking my app because these are not valid slugs. In fact, I think there should be no static time slugs being passed in at all. 

I have deployed a similar app, https://github.com/sanity-io/sanity-template-nextjs-appdir-personal-website, to Vercel and it works just fine, which is driving me crazy.

**Comment by amyegan** [comment]
Thanks for reaching out! This may have something to do with your configuration, possibly a build setting. Can you share your repo or a minimal reproduction?

I think it would also be worth sharing this in https://github.com/vercel/next.js/discussions/41745 where the team is collecting `app` directory feedback

**Reply by ghost** [reply]
Thank you for your time and attention! I shared this discussion link in the app directory feedback chain and greatly appreciate everybody working on this (amazing) new technology. I also made my repository public at https://github.com/sharpsteelsoftware1/josephmalicke-com.

**Reply by sroussey** [reply]
If you want to be dynamic only then you need to add `export const dynamic = 'dynamic-only'` to you page. It is sending urlencoded version of “[slug]”. This happens if you are not sending the params for making static pages.

**Reply by ghost** [reply]
Is this documented anywhere? Somehow I must have missed it. I added it to page.tsx, pushed, but Vercel logs are still showing that it is sending urlencoded [slug]. In other words, no change. 

```
import { sanityItem } from "../../lib/sanity";
import { makeJsx } from "../../components/feed";
export const dynamic = "dynamic-only";
export default async function BlogPage({
  params,
}: {
  params: { slug: string };
}) {
  const rawItem = await sanityItem(params.slug);
  console.log(
    `Params are ${JSON.stringify(params)} and rawItem is ${JSON.stringify(
      rawItem
    )}`
  );

  // const component = makeJsx(rawItem);
  // return <>{makeJsx(false)(rawItem)}</>;
  return <div>{JSON.stringify(rawItem, null, 2)}</div>;
  // return component;
}


```





info  - Generating static pages (0/4)
16:43:21.417 | info  - Generating static pages (1/4)
16:43:21.452 | info  - Generating static pages (2/4)
16:43:21.828 | Params are {"slug":"%5Bslug%5D"} and rawItem is undefined
16:43:21.831 | TypeError: Cannot read properties of undefined (reading '_type')
16:43:21.831 | at makeJsx (/vercel/path0/.next/server/chunks/350.js:295:20)
16:43:21.831 | at BlogPage (/vercel/path0/.next/server/app/[slug]/page.js:300:89)
16:43:21.831 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
16:43:21.847

<br class="Apple-interchange-newline">

**Reply by ghost** [reply]
Just found https://beta.nextjs.org/docs/api-reference/segment-config that talks about forcing dynamic rendering. Even when using on page.tsx

export const dynamic = "force-dynamic";
export const dynamicParams = true;

there is still no change and nextjs is building it statically and passing in /slug/

**Comment by ghost** [comment, marked as answer]
I updated to "next": "v13.1.7-canary.6", and it solved the issue
